sale_order_import_edifact/wizard/sale_order_import.py

```diff
@@ -0,0 +1,248 @@
+import json
+import logging
+import os
+from base64 import b64decode
+from collections import defaultdict
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+logger = logging.getLogger(__name__)
+
+
+class SaleOrderImport(models.TransientModel):
+    _inherit = "sale.order.import"
+
+    import_type = fields.Selection(
+        selection_add=[("edifact", "EDIFACT")], ondelete={"edifact": "cascade"}
+    )
+    state = fields.Selection(
+        selection_add=[("tech", "Technical")], ondelete={"tech": "set default"}
+    )
+    parsed_json = fields.Text()
+    edifact_ok = fields.Boolean(compute="_compute_edifact_ok", store=True)
+
+    @api.depends("order_file", "order_filename")
+    def _compute_edifact_ok(self):
+        for rec in self:
+            if rec.order_file and rec.order_filename:
+                conf_ext = self.env.context.get("file_ext", "txt,d96a")
```

Comment:
Instead of depending purely on the ctx, I think it's better to have a field and via the ctx we set its default.